src/aspire/image/__init__.py

```diff
@@ -78,6 +79,53 @@ def _im_translate2(im, shifts):
     return im_translated
 
 
+def normalize_bg(imgs, bg_radius=1.0, do_ramp=True):
+    """
+    Normalize backgrounds and apply to a stack of images
+
+    :param imgs: A stack of images in L-by-L-by-N array
+    :param bg_radius: Radius cutoff to be considered as background (in image size)
+    :param do_ramp: When it is `True`, fit a ramping background to the data
+            and subtract. Namely perform normalization based on values from each image.
+            Otherwise, a constant background level from all images is used.
+    :return: The modified images
+    """
+    L = imgs.shape[1]
+    grid = grid_2d(L)
+    mask = (grid['r'] > bg_radius)
+
+    if do_ramp:
+        # Create matrices and reshape the background mask
+        # for fitting a ramping background
+        ramp_mask = np.vstack((grid['x'][mask].flatten(),
+                           grid['y'][mask].flatten(),
+                           np.ones(grid['y'][mask].flatten().size))).T
+        ramp_all = np.vstack((grid['x'].flatten(), grid['y'].flatten(),
+                          np.ones(L*L))).T
+        mask_reshape = mask.reshape((L*L))
+        imgs = imgs.reshape((L*L, -1))
+
+        # Fit a ramping background and apply to images
+        coeff = lstsq(ramp_mask, imgs[mask_reshape])[0]
+        imgs = imgs - ramp_all @ coeff
+        imgs = imgs.reshape((L, L, -1))
+
+    # Apply mask images and calculate mean and std values of background
+    imgs_masked = (imgs * np.expand_dims(mask, 2))
+    denominator = np.sum(mask)
+    first_moment = np.sum(imgs_masked, axis=(0, 1))/denominator
+    second_moment = np.sum(imgs_masked ** 2, axis=(0, 1))/denominator
+    mean = first_moment
+    variance = second_moment - mean**2
+    std = np.sqrt(variance)
+
+    # Normalize the images by background
+    imgs -= mean
+    imgs /= std
+
+    return imgs
```

Comment:
We should avoid creating variables which are only used once. We could have saved 5 lines by simply saying:
```
return (imgs - mean) / np.sqrt(variance)
```

One could remove the variables `mean` and `variance` altogether and just deal with `first_moment` and `second_moment` but that might be stretching it a bit too far for the casual reader.

Reply:
Yes. Changed in the new commit db7149e.